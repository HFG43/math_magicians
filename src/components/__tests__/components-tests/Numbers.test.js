import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import NumberPad from '../../numbers';

it('Renders <NumberPad /> correctly', () => {
  const handleClick = jest.fn();

  const numberPad = renderer
    .create(<NumberPad handleClickNumber={handleClick} />)
    .toJSON();
  expect(numberPad).toMatchSnapshot();
});

it('Should be call "5" when button 5 is clicked', () => {
  const handleClick = jest.fn();
  const { getByText } = render(<NumberPad handleClickNumber={handleClick} />);
  fireEvent.click(getByText('5'));
  expect(handleClick).toHaveBeenCalledWith('5');
});
