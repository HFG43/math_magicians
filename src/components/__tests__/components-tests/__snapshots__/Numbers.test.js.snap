// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <NumberPad /> correctly 1`] = `
<div
  className="calc_num_pad"
>
  <span>
    <button
      id="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
  </span>
  <span>
    <button
      id="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
  </span>
  <span>
    <button
      id="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
  </span>
  <span>
    <button
      id="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
  </span>
  <span>
    <button
      id="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
  </span>
  <span>
    <button
      id="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
  </span>
  <span>
    <button
      id="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
  </span>
  <span>
    <button
      id="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
  </span>
  <span>
    <button
      id="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
  </span>
  <span>
    <button
      id="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
  </span>
  <span>
    <button
      id="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
  </span>
</div>
`;

exports[`Renders <NumberPad /> correctly 2`] = `
<div
  className="calc_num_pad"
>
  <span>
    <button
      id="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
  </span>
  <span>
    <button
      id="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
  </span>
  <span>
    <button
      id="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
  </span>
  <span>
    <button
      id="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
  </span>
  <span>
    <button
      id="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
  </span>
  <span>
    <button
      id="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
  </span>
  <span>
    <button
      id="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
  </span>
  <span>
    <button
      id="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
  </span>
  <span>
    <button
      id="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
  </span>
  <span>
    <button
      id="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
  </span>
  <span>
    <button
      id="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
  </span>
</div>
`;
